### ISSUE TITLE
Sunstone alloy and Moonlight hide researches incompletable

### ISSUE THREAD

**Issue by ChaoticLaughter** [AirBurn0/am2playground#13]
When trying to get the research papers in the thaumonomicon, it will tell you that you gained the paper but won't actually give you one, consuming the paper like normal.
<img width="319" alt="image" src="https://github.com/AirBurn0/am2playground/assets/49538589/8d201dee-5c23-433b-b894-bfa2bc552e4c">
![image](https://github.com/AirBurn0/am2playground/assets/49538589/a5b64453-386a-446d-b629-efa9ec5b87fc)
![image](https://github.com/AirBurn0/am2playground/assets/49538589/8ad5aaf1-f0f2-4564-b410-7c52662ccb18)

**Comment by AirBurn0** [comment]
Oh...
It should automatically give you research completion without any notes and doesn't require paper...
Okay, sounds wierd, will test it out today.

**Comment by ChaoticLaughter** [comment]
yeah it's on v0.11.3, I don't know what's going on, the magic material researches appeared when I scanned the respective items, but it acted like it needed to be properly researched

**Comment by AirBurn0** [comment]
> Edit2: Thaumcraft's hard research difficulty is the culprit. (you must research _everything_ via table.)

Yup, you are 110% right. 

Originally, you should have simply picked this study at the cost of 0 per each aspect (cuz I really hate unnecessary researches and waste of aspects for nothing). I thought it would be funny.
But in reality, at the hard difficulty level, will be no peaceful solution, and it tries to find the aspect with the highest cost. Search algorithm is written in such a way that everything less than zero is a joke for it (base checking cost is basically 0, and to be counted, aspect cost must be greater than 0).
So, it just fails, return null, gives nothing and dies from the cringe.

**TLDR**: Will be fixed in next upd, thanks you for reporting this, I would never have found this on my own.
If you need a version with a fix ASAP, I can quickly compile a dev build, but there's a lot of unfinished content here which at this stage looks like it's useless and sometimes have 0 gameplay sense.

**Comment by ChaoticLaughter** [comment]
I don't need a fix, I bumped the config from hard to normal, feels a lot better that way anyways not having to do the minigame for literally everything. 

Unrelated: The illusion cap research is fucking hell, me and a friend who is much better than me couldn't figure it out so I decided to use gadomancy's auto-research item and a lot of cognitio

**Comment by AirBurn0** [comment]
> Unrelated: The illusion cap research is fucking hell, me and a friend who is much better than me couldn't figure it out so I decided to use gadomancy's auto-research item and a lot of cognitio
Yeah that's kinda the point!
I'm glad you liked it! I specifically tried to design them in the most buttcracking way :)
